src/libexpr/eval.cc

```diff
@@ -1943,57 +1946,58 @@ std::optional<string> EvalState::tryAttrsToString(const Pos & pos, Value & v,
     return {};
 }
 
-string EvalState::coerceToString(const Pos & pos, Value & v, PathSet & context,
-    bool coerceMore, bool copyToStore, bool canonicalizePath)
+std::string_view EvalState::coerceToString(const Pos & pos, Value & v, PathSet & context,
```

Comment:
Returning a `string_view` here is too ugly/risky for me, since it requires the caller to manage a buffer. If only C++ had Rust's Cow type...

Reply:
yeah, that'd be great. i guess we could add a simple variant of that?

Reply:
Hm yes, maybe a (wrapper around) `std::variant<std::string_view, std::string>` would be pretty nice.